main.go

```diff
@@ -515,56 +523,67 @@ func createLogParserValidator(client client.Client) *logparserwebhook.Validating
-		OverrideConfigMap: types.NamespacedName{Name: overrideConfigMapName, Namespace: telemetryNamespace},
-		Service: collectorresources.ServiceConfig{
-			OTLPServiceName: "telemetry-otlp-traces",
+	config := tracepipeline.Config{
+		Gateway: gateway.Config{
+			Namespace: telemetryNamespace,
+			BaseName:  "telemetry-trace-collector",
+			Deployment: gateway.DeploymentConfig{
+				Image:                traceCollectorImage,
+				PriorityClassName:    traceCollectorPriorityClass,
+				BaseCPULimit:         resource.MustParse(traceCollectorCPULimit),
+				DynamicCPULimit:      resource.MustParse(traceCollectorDynamicCPULimit),
+				BaseMemoryLimit:      resource.MustParse(traceCollectorMemoryLimit),
+				DynamicMemoryLimit:   resource.MustParse(traceCollectorDynamicMemoryLimit),
+				BaseCPURequest:       resource.MustParse(traceCollectorCPURequest),
+				DynamicCPURequest:    resource.MustParse(traceCollectorDynamicCPURequest),
+				BaseMemoryRequest:    resource.MustParse(traceCollectorMemoryRequest),
+				DynamicMemoryRequest: resource.MustParse(traceCollectorDynamicMemoryRequest),
+			},
+			Service: gateway.ServiceConfig{
+				OTLPServiceName: "telemetry-otlp-traces",
+			},
 		},
-		MaxPipelines: maxTracePipelines,
+		OverridesConfigMapName: types.NamespacedName{Name: overridesConfigMapName, Namespace: telemetryNamespace},
+		MaxPipelines:           maxTracePipelines,
 	}
 	overridesHandler := overrides.New(configureLogLevelOnFly, &kubernetes.ConfigmapProber{Client: client})
 
 	return telemetrycontrollers.NewTracePipelineReconciler(
 		client,
-		tracepipelinereconciler.NewReconciler(client, config, &kubernetes.DeploymentProber{Client: client}, overridesHandler),
+		tracepipeline.NewReconciler(client, config, &kubernetes.DeploymentProber{Client: client}, overridesHandler),
 	)
 }
 
 func createMetricPipelineReconciler(client client.Client) *telemetrycontrollers.MetricPipelineReconciler {
-	config := collectorresources.Config{
-		Namespace: telemetryNamespace,
-		BaseName:  "telemetry-metric-gateway",
-		Deployment: collectorresources.DeploymentConfig{
-			Image:                metricGatewayImage,
-			PriorityClassName:    metricGatewayPriorityClass,
-			BaseCPULimit:         resource.MustParse(metricGatewayCPULimit),
-			DynamicCPULimit:      resource.MustParse(metricGatewayDynamicCPULimit),
-			BaseMemoryLimit:      resource.MustParse(metricGatewayMemoryLimit),
-			DynamicMemoryLimit:   resource.MustParse(metricGatewayDynamicMemoryLimit),
-			BaseCPURequest:       resource.MustParse(metricGatewayCPURequest),
-			DynamicCPURequest:    resource.MustParse(metricGatewayDynamicCPURequest),
-			BaseMemoryRequest:    resource.MustParse(metricGatewayMemoryRequest),
-			DynamicMemoryRequest: resource.MustParse(metricGatewayDynamicMemoryRequest),
+	config := metricpipeline.Config{
+		Agent: agent.Config{
+			Namespace: telemetryNamespace,
+			BaseName:  "telemetry-metric-agent",
+			DaemonSet: agent.DaemonSetConfig{
+				Image: traceCollectorImage,
```

Comment:
Set same priority class as for Fluent Bit

Reply:
Done